src/app/shared/utils/hash-helper.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export class HashHelper {
+
+  public static hashString(str: string): number {
+    let hash = 0;
+
+    if (str.length === 0) {
+      return hash;
+    }
+
+    for (let i = 0; i < str.length; i++) {
+      const character = str.charCodeAt(i);
+      hash = ((hash << 5) - hash) + character;
+      hash = hash & hash; // Convert to 32bit integer
+    }
```

Comment:
Seems like a solution copied from Stack Overflow. But there was also the [ES6 version](https://stackoverflow.com/a/34842797/3047439). ;)

Reply:
Since it happens often, I suggest keeping it this way, as this aproach should still be faster on V8.